cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -4,7 +4,64 @@
+#[[
+The purpose of this machinery is to search Python and set both the ROOT and PyROOT related
+Python variables.
+
+ROOT:
+    - PYTHON_EXECUTABLE (set when Interpreter is found)
+    - PYTHON_VERSION_STRING        ""
+    - PYTHON_VERSION_MAJOR        ""
+    - PYTHON_VERSION_MINOR        ""
+
+PyROOT: a set of specific variables is needed for all the Python versions used to build PyROOT.
+At the time of writing we only have two, called simply Main and Other.
+
+    - PYTHON_EXECUTABLE_Development_Main (set when Interpreter is found)
+    - PYTHON_VERSION_STRING_Development_Main        ""
+    - PYTHON_VERSION_MAJOR_Development_Main        ""
+    - PYTHON_VERSION_MINOR_Development_Main        ""
+    - PYTHON_INCLUDE_DIRS_Development_Main (set when Development is found)
+    - PYTHON_LIBRARIES_Development_Main        ""
+    - PYTHON_UNDER_VERSION_STRING_Development_Main (used to give a version-dependent name to the libraries, to allow
+    multiple builds)
+    - Development_Python${version}_FOUND (set if both Interpreter and Development are found for the specified ${version})
+
+    - PYTHON_EXECUTABLE_Development_Other (set when Interpreter is found)
+    - PYTHON_VERSION_STRING_Development_Other        ""
+    - PYTHON_VERSION_MAJOR_Development_Other        ""
+    - PYTHON_VERSION_MINOR_Development_Other        ""
+    - PYTHON_INCLUDE_DIRS_Development_Other (set when Development is found)
+    - PYTHON_LIBRARIES_Development_Other        ""
+    - PYTHON_UNDER_VERSION_STRING_Development_Other (used to give a version-dependent name to the libraries, to allow
+    multiple builds)
+    - Development_Python${version}_FOUND (set if both Interpreter and Development are found for the specified ${version})
+
+^^^^^^^^
+
+Explanation of the machinery:
+
+The first distinction is based on the CMake version used to build. If it is >= 3.14, than PyROOT can be built
+for multiple Python versions. In case CMake >= 3.14, then we check if PYTHON_EXECUTABLE was specified by the user;
+if so, PyROOT is built with only that version, otherwise an attempt to build it both with Python2 and 3 is done.
+
+If PYTHON_EXECUTABLE is specified:
+    - we check which version it is (from here call X) and call the relative
+    find_package(PythonX COMPONENTS Interpreter Development Numpy)
+    - if Interpreter is found, we set the ROOT related Python variables
+    - if Development is found, we set the PyROOT_Main variables (+ Numpy ones)
+
+If PYTHON_EXECUTABLE is NOT specified:
+    - we look for Python3, since we want the highest version to be the preferred one
+    - if Python3 Interpreter is found, we set the ROOT related Python variables
+    - if Python3 Development is found, we set the PyROOT_Main variables (+ Numpy ones)
+    - we then look for Python2, requiring it to be >= 2.7:
+        - if Python3 Interpreter wasn't previously found, then it means that Python2 becomes the one used to
```

Comment:
This is actually happening in two steps: `- if Python3 Interpreter wasn't previously found, and Python2 Interpreter is found, ROOT will be built with Python2, so we set the corresponding ROOT variables. If, in addition, Python2 Development is found, we will build PyROOT and PyMVA with Python2 (it becomes the Main one for PyROOT) and we set the Development_Main variables - plus the NumPy ones, if NumPy is found.`